### PR TITLE
Documentation for Azure Monitor scaler

### PR DESCRIPTION
**Pull request by melmaliacone**
Documentation for the Azure Monitor scaler. I wasn't sure what to put for the `Availability`.

### CODE REVIEW

File: content/scalers/azure-monitor.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
++++
+fragment = "content"
+weight = 100
+title = "Azure Monitor"
+background = "light"
++++
+
+Scale applications based on Azure Monitor metrics.
+
+**Availability:** ?? | **Maintainer:** Community
```
Comment (jeffhollan) [inline content/scalers/azure-monitor.md]:
```suggestion
**Availability:** 1.3 | **Maintainer:** Community
```

Comment (melmaliacone) [reply content/scalers/azure-monitor.md]: Done!

File: content/scalers/azure-monitor.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+The `activeDirectoryClientId` value is the name of the environment variable your deployment uses to get the active directory client id. The `activeDirectoryClientPassword` value is the name of the environment variable your deployment uses to get the active directory client password. This is usually resolved from a `Secret V1` or a `ConfigMap V1` collections. `env` and `envFrom` are both supported.
+
+**Parameter list:**
+
+- `resourceURI` is the shortened URI to the Azure resource. The format is `"<resourceProviderNamespace>/<resourceType>/<resourceName>"`. Required. 
+- `tenantId` is the tenant id for the Azure resource. Used for authentication. Required.
+- `subscriptionId` is the subscription id for the Azure resource. Used for determining the full resource URI. Required.
+- `resourceGroupName` is the resource group for the Azure resource. Required.
+- `metricName` is the name of the metric. Must be an officially supported metric from https://docs.microsoft.com/en-us/azure/azure-monitor/platform/metrics-supported. Required.
```
Comment (tomkerkhove) [inline content/scalers/azure-monitor.md]:
```suggestion
- `metricName` is the name of the metric. Must be an officially supported metric  found in the [official documentation](https://docs.microsoft.com/en-us/azure/azure-monitor/platform/metrics-supported). Required.
```

File: content/scalers/azure-monitor.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+- `metricFilter` is used to define a more specific part of the resource. You can filter by supported dimensions of the metric found in https://docs.microsoft.com/en-us/azure/azure-monitor/platform/metrics-supported. Optional.
```
Comment (tomkerkhove) [inline content/scalers/azure-monitor.md]:
```suggestion
- `metricFilter` is used to define a more specific part of the resource. You can filter by supported dimensions of the metric found in the [official documentation](https://docs.microsoft.com/en-us/azure/azure-monitor/platform/metrics-supported). Optional.
```

File: content/scalers/azure-monitor.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+- `metricAggregationInterval` is the collection time of the metric. Reported in the format `"mm:hh:ss"`. The default value is `"0:5:0"` (5 minutes). Optional.
+- `metricAggregationType` is the aggregation method of the Azure monitor metric. Some possible values include Average, Total, Maximum. Based on what is supported in https://docs.microsoft.com/en-us/azure/azure-monitor/platform/metrics-supported. No default. Required.
```
Comment (tomkerkhove) [inline content/scalers/azure-monitor.md]:
```suggestion
- `metricAggregationType` is the aggregation method of the Azure monitor metric. Some possible values include `Average`, `Total`, `Maximum` with a full list in the [official documentation](https://docs.microsoft.com/en-us/azure/azure-monitor/platform/metrics-supported). No default. Required.
```

File: content/scalers/azure-monitor.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+      resourceGroupName: azureMonitor
+      metricName: kube_pod_status_ready
+      metricFilter: namespace eq 'default'
+      metricAggregationInterval: "0:1:0"
+      targetValue: "1"
+      activeDirectoryClientId: CLIENT_ID_ENV_NAME
+      activeDirectoryClientPassword: CLIENT_PASSWORD_ENV_NAME
+```
+
+The `activeDirectoryClientId` value is the name of the environment variable your deployment uses to get the active directory client id. The `activeDirectoryClientPassword` value is the name of the environment variable your deployment uses to get the active directory client password. This is usually resolved from a `Secret V1` or a `ConfigMap V1` collections. `env` and `envFrom` are both supported.
```
[the remark this answers is not in the document]
Comment (melmaliacone) [reply content/scalers/azure-monitor.md]: Good call. Will do

[the remark this answers is not in the document]
Comment (tomkerkhove) [reply content/scalers/azure-monitor.md]: Thanks!

File: content/scalers/azure-monitor.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+### Authentication Parameters
+
+You can use `TriggerAuthentication` CRD to configure the authentication by providing a set of Azure Active Directory credentials.
+
+**Credential based authentication:**
+
+- `activeDirectoryClientId` - Active Directory client id.
+- `activeDirectoryClientPassword` - Active Directory client password.
+
+The user will need access to read data from the Azure resource.
```
[the remark this answers is not in the document]
Comment (melmaliacone) [reply content/scalers/azure-monitor.md]: @tomkerkhove Added the details in the most recent commit. Let me know if you think it should be rephrased.

[the remark this answers is not in the document]
Comment (tomkerkhove) [reply content/scalers/azure-monitor.md]: LGTM, thanks!

**Comment by tomkerkhove** [comment]
Thanks @melmaliacone!

**Pull request merged**